scrape/scrape.go

```diff
@@ -980,6 +981,28 @@ func newScrapeCache() *scrapeCache {
 	}
 }
 
+func (c *scrapeCache) free() {
+	if c == nil {
+		return
+	}
+
+	for k := range c.series {
+		delete(c.series, k)
```

Comment:
I was reading the code for the cache a little bit more. It seems like the cache might shared between scrapeLoops. Not sure if this would delete other series that are currently in used.

I wonder if we should only be deleting the series associated with the deleted scrapeLoop?